src/NServiceBus.RabbitMQ.AcceptanceTests/ConfigureEndpointRabbitMQTransport.cs

```diff
@@ -76,13 +70,31 @@ ConnectionFactory CreateConnectionFactory()
     {
         var connectionFactory = new ConnectionFactory
         {
-            UserName = connectionStringBuilder["username"].ToString(),
-            Password = connectionStringBuilder["password"].ToString(),
-            VirtualHost = connectionStringBuilder["virtualhost"].ToString(),
-            HostName = connectionStringBuilder["host"].ToString(),
             AutomaticRecoveryEnabled = true
         };
 
+        object value;
+
+        if (connectionStringBuilder.TryGetValue("username", out value))
+        {
+            connectionFactory.UserName = value?.ToString();
```

Comment:
So yeah, I completely forgot about the `ConnectionFactory` defaults. For many of the values, `ConnectionConfiguration` has the same default values as `ConnectionFactory`, so this is fine here. 

Looking at the `?.` operators, I'm wondering how likely it would be to actually get a null value out of the connection builder at this point, but I don't suppose it hurts anything.

So, at this point, I'd say this looks good and disregard my comments around defaults!